packages/basic-crawler/src/internals/basic-crawler.ts

```diff
@@ -1115,6 +1140,18 @@ export class BasicCrawler<Context extends CrawlingContext = BasicCrawlingContext
         return isRequestListFinished && isRequestQueueFinished;
     }
 
+    private async _rotateSession(crawlingContext: Context) {
+        const { request } = crawlingContext;
+
+        if ((request.sessionRotationCount ?? 0) >= this.maxSessionRotations) {
+            throw new Error(`Session failed ${request.sessionRotationCount} times. `
+                + 'This might be caused by a misconfigured proxy or an invalid session pool configuration.');
```

Comment:
this might be a bit misleading, I just realized we have another limit on the session itself - there is the `maxErrorScore` which is basically the same thing on another level. in other words, with the defaults (`maxErrorScore: 3`) we will rotate the session anyway, at least I think it works this way. that means that a specific session didn't fail 10 times, it's more like 4 different sessions failed or the given request. 

i dont think we should change anything in the implementation, but reading the error message I felt like it's not 100% correct. 

maybe it's fine this way too, not a big deal.

Reply:
Good point, I can't really figure out any better wording for this... Maybe "Session got hard-rotated (x) times" or "Too many session errors"? Any of these are still applicable to the `maxErrorScore` issue too, really 🤷🏼

Reply:
maybe "Request failed because of proxy-related errors N times."?